src/stm32l0.c

```diff
@@ -0,0 +1,1076 @@
+     will.  Changing the PC to the start of RAM and single stepping
+     will immediately HardFault.
+
+     The stub functions can be both faster and simpler because they
+     have direct access to the MCU.  So, to permit stub operation in
+     the best circumstances, the NVM operation functions will check
+     the MCU state and either execute the stub or non-stub version
+     accordingly. The user can override stubs as well with a command
+     in case the detection feature fails.
+
+   o Erase would be more efficient if we checked for non-blank-ness
+     before initiating an erase.  This would have to be done in a stub
+     for efficiency.
+
+   o Mass erase broken.  The method for performing a mass erase is to
+     set the options for read protection, reload the option bytes, set
+     options for no protection, and then reload the option bytes
+     again.  The command fails because we lose contact with the target
+     when we perform the option byte reload.  For the time being, the
+     command is disabled.
+
+   o Errors.  We probably should clear SR errors immediately after
+     detecting them.  If we don't then we always must wait for the NVM
+     module to complete the last operation before we can start another.
+
+   o There are minor inconsistencies between the stm32l0 and the
+     stm32l1 in when handling NVM operations.
+
+     o When we erase or write individual words (not half-pages) on the
+       stm32l0, we set the PROG bit.  On the stm32l1 the PROG bit is
+       only set when erasing.  This  is not documented in the register
+       summaries, but in the functional quick reference.
+
+     o On the STM32L1xx, PECR can only be changed when the NVM
+       hardware is idle.  The STM32L0xx allows the PECR to be updated
+       while an operation is in progress.
+
+   o Performance.  The throughput for writing is not high.  We suspect
+     it may be possible to improve throughput significantly by increasing
+     the MCU clock.  The code, as is, offers a simplicity without
+     undue knowledge of the inner workings of the MCUs.  Increasing
+     clock frequencies would require substantial knowledge of the
+     clock tree.
+
+  0x1ff80000: 0x00aa 0xff55 OK
+  0x1ff80004: 0x8070 0x7f8f OK
+  0x1ff80008: 0x0000 0xffff OK
+  OPTR: 0x807000aa, RDPROT 0, WPRMOD 0, WDG_SW 1, BOOT1 1
+
+
+    Options code
+p *(int*)0x40022004 = 1
+p *(int*)0x4002200c = 0x89abcdef
+p *(int*)0x4002200c = 0x02030405
+p *(int*)0x40022014 = 0xfbead9c8
+p *(int*)0x40022014 = 0x24252627
+p *(int*)0x40022004 = 0x10
+x/4wx 0x40022000
+p *(int*)0x1ff80000 = 0xff5500aa
+
+  l1 program
+p *(int*)0x40023c04 = 1
+p *(int*)0x40023c0c = 0x89abcdef
+p *(int*)0x40023c0c = 0x02030405
+p *(int*)0x40023c10 = 0x8c9daebf
+p *(int*)0x40023c10 = 0x13141516
+
+
+p *(int*)0x40023c14 = 0xfbead9c8
+p *(int*)0x40023c14 = 0x24252627
+
+*/
+
+#define CONFIG_STM32L1          /* Include support for STM32L1 */
```

Comment:
That depends on you.  If you want to replace the existing L1 support, then I'll make it permanent.  If you don't I'd prefer to leave the conditional in and work toward coalescing all of the Lx support in one place.